src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_CommitUniqueCompletionListItem.cs

```diff
@@ -73,17 +73,17 @@ private void CommitUniqueCompletionListItemAsynchronously()
             // We're kicking off async work.  Track this with an async token for test purposes.
             var token = ((IController<Model>)this).BeginAsyncOperation(nameof(CommitUniqueCompletionListItemAsynchronously));
 
-            var task = currentTask.ContinueWith(t =>
+            var task = currentTask.ContinueWith(async t =>
             {
-                this.AssertIsForeground();
+                await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
❕ This needs to yield if called from the main thread in order to preserve the original behavior.